server/app/api/v1/query_executor.py

```diff
@@ -0,0 +1,58 @@
+from fastapi import APIRouter, Depends, status
+from fastapi.responses import JSONResponse
+from sqlalchemy.orm import Session
+from app.api.v1.dependencies import get_current_user, get_db
+from app.crud.crud_user_database import crud_user_database
+from sqlalchemy import create_engine
+from fastapi.encoders import jsonable_encoder
+from sqlalchemy.sql import text
+from app.services.query_service import query_service 
+import logging
+
+router = APIRouter()
+logger = logging.getLogger("mql")
+
+@router.get("/sql-data")
```

Comment:
I believe this should be a `POST` request because we are not fetching a resource that is already defined on the server. We are "executing" a query (as the name implies), which is an action.

cc: @ashok-shuru

Reply:
It should be GET, because we are not updating any resource on our system. POST is generally used when we are updating something. @harshshurutech

Reply:
@ashok-shuru  GraphQL queries are sent as `POST` requests even when we are trying to fetch data and not update/submit new data because:

1. POST allows for sending large and complex query structures in the request body.
2. POST requests are not cached by default by intermediary systems like CDNs or proxies. Since GraphQL queries can contain sensitive or personalized information, avoiding caching by default can be a security advantage.

While it may seem counterintuitive to use a POST request in this case, the benefits cannot be ignored. Therefore, we can apply the same approach for executing queries in MQL as well.